src/ast/mod.rs

```diff
@@ -147,15 +150,121 @@ impl fmt::Display for Ident {
 }
 
 /// A name of a table, view, custom type, etc., possibly multi-part, i.e. db.schema.obj
-#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+///
+/// # Semantic Equality
+///
+/// This implements equals and hash in terms of the ident *values*, not
+/// including quotes.
```

Comment:
Eek! I suppose this is alright, but it's the first time (I think) that the SQL parser has been concerned with semantic equality. In general I would like the SQL parser to not attempt to be in that business—it's impossible—but this change seems targeted enough to be worth it.